apps/nextjs-website/src/lib/api.ts

```diff
@@ -1,12 +1,151 @@
-/** This file contains all the functions useful to get data from external resources */
-
-// Return all the pages
-export const getAllPages = async () => ({
-  // for now hardcoded, then retrieved from the cms
-  pages: [
-    { slug: ['a'] },
-    { slug: ['a', 'b'] },
-    { slug: ['a', 'b', 'c'] },
-    { slug: ['a', 'b', 'c', 'd'] },
-  ],
+import { z } from 'zod';
+
+// Zod schemas for validation
```

Comment:
Do we need all these properties? If not can we just define the properties we need?

Reply:
yes we need them all due to datas we'll get from the API calling. Without it, we cannot validate what we'll get and if there's some mismatch could be dangerous. Due to types needs to be declared we have only interface (or types) or zod method. In any of those cases, all those declarations are needed even if they seems to be a lot, 'cause the datas we're going to receive from the CMS are structured in this way and I don't suggest to avoid the data check.